prereise/gather/winddata/hrrr/grib.py

```diff
@@ -0,0 +1,117 @@
+class GribRecordInformation:
+    """Class to hold metadata on a grib record. grib (General Regularly-distributed
+    Information in Binary) is a concise data format commonly used in meteorology to
+    store historical and forecast weather data. grib files can have corresponding
+    index files that holds metadata about the data inside grib files. More info
+    on index files can be found at
+    `this link <https://github.com/blaylockbk/Herbie/blob/18945e4c5103386c98d08dcb2de590e2ac14c3d5/docs/user_guide/grib2.rst#how-grib-subsetting-works-in-herbie>`_
+
+    :param str message_number: message number of the grib record
+    :param str beginning_byte: byte that the grib record starts at
+    :param str ending_byte: byte that the grib record ends at. If None
+        then ending_byte is the end of file
+    :param str initialization_date: date and hour that the data is for
+    :param str variable: string designating what kind of data
+    :param str level: additional string that designates more information about
+        type of data
+    :param str forecast: designates how many hours into the future
+        this data is forecasted for
+    """
+
+    def __init__(
```

Comment:
Also optional, but this might be a good use case for a [dataclass](https://docs.python.org/3/library/dataclasses.html). Other than saving some boilerplate, it could also allow simplify things like comparing two instances, customizing the `repr`, etc.

Reply:
👍